proxy.c

```diff
@@ -206,7 +215,7 @@ void handle_client(int client_sock, struct sockaddr_in client_addr)
 {
     if ((remote_sock = create_connection()) < 0) {
         perror("Cannot connect to host");
-        return;
+        goto out;
```

Comment:
I'm sorry, I missed the context. If it's within one function than let it stay here, I'd appreciate if you could just name it cleanup: to make it more explicit what it does.